src/lib.rs

```diff
@@ -0,0 +1,29 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2022
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use pyo3::prelude::*;
+use pyo3::wrap_pyfunction;
+use pyo3::Python;
+
+mod edge_collections;
+mod nlayout;
+mod stochastic_swap;
+
+use stochastic_swap::*;
+
+#[pymodule]
+fn stochastic_swap_rs(_py: Python<'_>, m: &PyModule) -> PyResult<()> {
+    m.add_class::<edge_collections::EdgeCollection>()?;
```

Comment:
To check: are the `?` monadic short-circuits to return `Error` (or whatever the opposite of `Ok` is) from the scope?

Reply:
Yep, if `m.add_class()` returns an `Err` here the `?` will just short circuit return it here. It's basically syntactic sugar for

```rust
match m.class::<edge_collections::EdgeCollection>() {
    Ok(v) => v,
    Err(e) => return Err(e),
};
```

but using `?` makes propagating errors much simpler